scipy/optimize/_trustregion_exact.py

```diff
@@ -207,11 +207,12 @@ class IterativeSubproblem(BaseQuadraticSubproblem):
     # in formula 7.3.14 (p. 190) named as "theta".
     # As recommended there it value is fixed in 0.01.
     UPDATE_COEFF = 0.01
+    MAXITER_DEFAULT = 0  # zero means infinite.
```

Comment:
perhaps even another default is appropriate, but I don't know what's reasonable, 10, 100, 1000? Or set it equal to maxiter of the "outer" problem (or a multiple thereof)?

If we keep it at zero (infinite) or choose a large value, then perhaps a warning should be issued (once?) in the while loop for say `self.niter == 100` or so, which informs the user that many iterations are being performed here, and that there exists an option to adjust the maximum number of iterations used in the subproblem.